### PR TITLE
WT-10301 Replace _table_runtime with a better data structure in Workgen 

### PR DESCRIPTION
**Pull request by sonar3ddy**

### CODE REVIEW

**Review by sulabhM — APPROVED** [review]

File: bench/workgen/workgen.cxx [hunk 1]
```diff
@@ -428,21 +428,15 @@ ContextInternal::ContextInternal()
 
 ContextInternal::~ContextInternal()
 {
-    if (_table_runtime != nullptr)
-        delete _table_runtime;
+    if (_table_runtime.size() != 0)
+        _table_runtime.resize(0);
```
Comment (sulabhM) [inline bench/workgen/workgen.cxx:432]: You don't need to resize to destroy the allocated memory. Actually resizing doesn't delete the vector's internally allocated memory. You can remove the destructor and just use an implicit one at this point, the destruction of ContextInternal will take care of cleanup.

**Review by sulabhM — APPROVED** [review]

**Comment by mm-ng** [comment]
I think there might still be a problem with this, as per failures in the PR patch.

**Comment by sonar3ddy** [comment]
> Details

Hi Monica,
Thanks, I will look into these!

File: bench/workgen/workgen.cxx [hunk 2]
```diff
@@ -416,33 +416,23 @@ Context::operator=(const Context &other)
...
-    if (_runtime_alloced < _tint_last) {
-        // The array references are 1-based, we'll waste one entry.
-        TableRuntime *new_table_runtime = new TableRuntime[_tint_last + 1];
-        for (uint32_t i = 0; i < _runtime_alloced; i++)
-            new_table_runtime[i + 1] = _table_runtime[i + 1];
-        delete _table_runtime;
-        _table_runtime = new_table_runtime;
-        _runtime_alloced = _tint_last;
+    if (_table_runtime.size() < _tint_last) {
+        _table_runtime.resize(_tint_last);
```
Comment (sulabhM) [inline bench/workgen/workgen.cxx:435]: Ah! The index references into the vector/array are 1-based. So you need
```
 _table_runtime.resize(_tint_last + 1);
```

**Review by mm-ng — APPROVED** [review]
LGTM

**Pull request merged**